### PR TITLE
refactor: site to doks

### PR DESCRIPTION
**Pull request by DannyS03**
## Context
- Migrates the existing IPLD content to use the [Hugo Doks theme](https://github.com/h-enk/doks).
> This will later be upgraded to the [PL docs-starter package](https://github.com/protocol/docs-starter) (based on the Doks theme).
- Design influenced on design proposal created by @juliaxbow on [Figma](https://www.figma.com/file/gQkvSQjbPV8QRmxSIqNcqu/IPLD-Docs-Site?node-id=0%3A1&t=BeKI2YqZWwum4fYk-0).
- Extensive lint-based updates.
- Added site to Plausible analytics.
- Part of #240

### To-Do
- [x] Fix search
- [x] Landing page
- [ ] Syntax highlighting for schemas

## Latest preview

Please view the latest Fleek preview [here](https://ipld.on.fleek.co).

### CODE REVIEW

File: content/docs/data-model/kinds.md [hunk 1]
```diff
@@ -152,29 +155,27 @@ Codecs generally treat bytes and strings differently.
...
 
 Links are concretely implemented as [`CID`](/glossary/#cid)s.
 
+## Implementation References
 
-Implementation References
--------------------------
-
-- Kinds in Go: https://github.com/ipld/go-ipld-prime/blob/master/kind.go
+Kinds in Go: https://github.com/ipld/go-ipld-prime/blob/master/kind.go
```
Comment (RangerMauve) [inline content/docs/data-model/kinds.md:181]: As per #255 

```suggestion
Kinds in Go: https://github.com/ipld/go-ipld-prime/blob/master/datamodel/kind.go
```

**Review by RangerMauve — COMMENTED** [review]
This is looking great! I appreciate that the most important links (to fixtures) are still working.

Getting rid of some of the mostly-empty index.md files is useful. The navigation in general feels a lot more approachable now.

**Comment by BigLep** [comment]
@DannyS03 : thanks for doing this!

Notes from 2022-11-29 triage: can you maybe convert your todos to a checklist so it's clear what's still remaining?  I want to make sure we give clear signal to maintainers on when this is fully ready for review and could be merged if they approved.

**Comment by DannyS03** [comment]
> @DannyS03 : thanks for doing this! IPLD maintainers really appreciate you getting this in before the end of the year!
> 
> Notes from 2022-11-29 triage: can you maybe convert your "To-Dos" to a checklist so it's clear what's still remaining? I want to make sure we give clear signal to maintainers on when this is fully ready for review and could be merged if they approved.
> 
> The main thing that @rvagg wants to see is the syntax highlighting for schemas.
> 
> @RangerMauve wants confirmation that the todos are handled.

Thanks and no problem!! Sure thing. 

Regarding the "landing page" to-do: the current landing page with Doks is a simplified version of the existing one. We can re-create the current one (or something similar). Or, the docs team will soon take some time to create new and sleek landing pages across the docs sites, which we can do for the IPLD site too.

Regarding the highlighting for schemas @rvagg, Doks supports highlight.js by default, would we be able to use the existing one?

**Comment by RangerMauve** [comment]
> Regarding the "landing page" to-do: the current landing page with Doks is a simplified version of the existing one. We can re-create the current one (or something similar)

Having a new sleek landing page is cool, I think the existing landing page in your PR is "good enough" as well. 👍

**Comment by rvagg** [comment]
I went looking for my previous excursions into ipldsch syntax highlighting because I thought I had highlight.js definitions for it somewhere, I found a file called `highlight.js` but it's a PrismJS definition:

* https://github.com/ipld/specs/blob/a7b9376ebd43aeabba7d78487db3d9df456b7714/.vuepress/highlight.js
* repeated in this repo, maybe updated a bit? https://github.com/ipld/ipld/blob/015ff80aea78f588b981c6cef7f1e9bd59d31f48/.site/prism-ipldsch.js
* also vscode definitions but those are a bit different: https://github.com/rvagg/vscode-ipldsch/tree/69cebb48a74c7e635de584cadecabef70585f939/syntaxes

I think highlight.js has its own format which is different again and I don't believe I have one for that yet!

**Comment by BigLep** [comment]
@DannyS03 : are you able to take on the syntax highlighting stuff?  Once we get that in, I think we're good to ship this.

**Comment by DannyS03** [comment]
@BigLep ah yes, will do!

**Comment by DannyS03** [comment]
Update: When configuring the syntax highlighting for schemas, I noticed that highlighting in general is not working; figuring that out.

<details>
<summary>Schema syntax highlighting for Highlight.js</summary>

```
const ipldsch = {
    case_insensitive: true,
    illegal: ':',
    contains: [
      {
        className: 'keyword',
        begin: /^[ \t]*(?:type|advanced)[ \t][A-Z](_?[A-Za-z0-9])*\b/m,
        contains: [
          {
            className: 'keyword',
            begin: /^[ \t]*(type|advanced)/m,
          },
          {
            className: 'class-name',
            begin: /[\w]+$/,
          },
        ],
      },
      {
        className: 'keyword',
        begin: /\b(?:bool|int|float|string|bytes|null|nullable|optional)\b/,
      },
      {
        className: 'builtin',
        begin: /\b(struct|union|enum)(?=[ \t]*\{)\b/,
      },
      {
        begin: /^}[ \t]representation\b/m,
        contains: [
          {
            className: 'builtin',
            begin: /representation/,
          },
        ],
      },
      {
        className: 'operator',
        begin: /=/,
      },
      {
        className: 'number',
        begin: /\b-?\d+\.?\d*(?:e[+-]?\d+)?\b/i,
      },
      {
        className: 'punctuation',
        begin: /[(){}:[\]|&]/,
      },
      {
        className: 'string',
        begin: /(")(?:\\[\s\S]|(?!\1)[^\\])*\1/,
        greedy: true,
      },
      {
        className: 'comment',
        begin: /(^|[^"])#.*/,
        lookbehind: true,
        greedy: true,
      },
    ],
  }

  module.exports = ipldsch
```

</details>

**Comment by johnnymatthews** [comment]
Bad news, the process for adding a new language is a bit more in depth from what I initially thought. The [Highlight.js team have a checklist that we need to run through](https://github.com/highlightjs/highlight.js/blob/main/extra/3RD_PARTY_QUICK_START.md) in order to add a 3rd party language into our project.

Following this checklist and adding a 3rd party language will likely take more than an hour. I can't spare that time right now, but I may be able to in a few weeks once the FVM project has stabilized (i.e. no longer requiring 90% of my attention).

I'd suggest setting any ` ```ipldsch ` codeblocks to ` ```plaintext ` for now, with an eye to addressing the syntax highlighting feature in the near future.

**Comment by DannyS03** [comment]
> Bad news, the process for adding a new language is a bit more in depth from what I initially thought. The [Highlight.js team have a checklist that we need to run through](https://github.com/highlightjs/highlight.js/blob/main/extra/3RD_PARTY_QUICK_START.md) in order to add a 3rd party language into our project.
> 
> Following this checklist and adding a 3rd party language will likely take more than an hour. I can't spare that time right now, but I may be able to in a few weeks once the FVM project has stabilized (i.e. no longer requiring 90% of my attention).
> 
> I'd suggest setting any `` ```ipldsch `` codeblocks to `` ```plaintext `` for now, with an eye to addressing the syntax highlighting feature in the near future.

The issue was more to do with highlighting "in general" not working. I realized when attempting to add the custom highlighting, which I intend to continue with.
> Update: When configuring the syntax highlighting for schemas, I noticed that highlighting in general is not working; figuring that out.

**Comment by johnnymatthews** [comment]
> The issue was more to do with highlighting "in general" not working.

Ah, missed that.

But it looks like it _is_ working. The codeblock under **Bytes** in this screenshot is being highlighted correctly:

Screen Shot 2023-01-17 at 12 48 18 PM

@DannyS03 can you link to a specific instance of codeblocks not being highlighted?

**Comment by DannyS03** [comment]
The example I shared previously:
> When I say "highlighting in general" is not working, it doesn't seem to recognize highlight.js or the default css. I am using the "hello world" doc just for testing code blocks (as seen in the screenshots). The {{< code-block language="ipldsch" >}} is custom, but you'll notice that the js code block doesn't highlight. As an example for comparison: https://docs.libp2p.io/guides/getting-started/javascript/#running-libp2p, https://raw.githubusercontent.com/libp2p/docs/master/content/guides/getting-started/javascript.md, which uses the same configuration (Doks default)

<div style="display: flex;">
  
  
</div>

**Comment by BigLep** [comment]
2023-01-24 IPLD triage conversation: 
We'd like to get the highlighting work because the concern is that if we don't do it now it's going to get dropped and not picked up given other priorities.
If we move forward with the doks migration as is, will someone still own the highlighting afterwards?

**Comment by johnnymatthews** [comment]
I'm still not sure what the issue actually is. I've asked @DannyS03 to create a video showing what's going on and exactly what the problem is.

**Comment by DannyS03** [comment]
> I'm still not sure what the issue actually is. I've asked @DannyS03 to create a video showing what's going on and exactly what the problem is.

@johnnymatthews https://www.loom.com/share/8c9d127ceef84e13a91c4b6b239a1259

**Comment by DannyS03** [comment]
@johnnymatthews did you get the chance to check the video out?

**Comment by DannyS03** [comment]
**Update**: Attempted to sandbox basic configuration for highlight.js and then figured out the issue I was facing. Highlighting should work fine and I've added (what should be) the correct configuration for the custom schema highlighting definition.

It is also being recognized (this is just a console.log of the definition):

image

However, the markdown code blocks don't seem to be recognizing the definition, and I am assuming it has to do with the definition being passed to the markdown parser.

**Comment by ElPaisano** [comment]
@BigLep can you add me as the assignee here? I don't have permission to do so.  I can probably take this and push it across the finish line, or at least see if someone else in docs can take it

**Comment by rvagg** [comment]
@ElPaisano go for it! But it might be best to fork from https://github.com/DannyS03/ipld/tree/refactor/migrate-to-doks and do another PR from there. I don't think there's much else we can do wrt permissions.

**Comment by rvagg** [comment]
I've pushed a branch here, refactor/migrate-to-doks, that should be this PR but rebased to master, if someone wants to pick this work up then they can use that as a base.

**Comment by johnnymatthews** [comment]
Nice, big thanks @rvagg! @ElPaisano you should now be able to work off https://github.com/ipld/ipld/tree/refactor/migrate-to-doks (instead of trying to push to `$this` fork.

**Comment by ElPaisano** [comment]
@rvagg @johnnymatthews  cheers thanks!

**Comment by BigLep** [comment]
@ElPaisano : I've set you as the assignee.  I'd love to get this one closed out given it's been open for ~5 months :)

**Comment by johnnymatthews** [comment]
Given the size of this PR, it's gonna take a significant amount of time to review. We'll discuss how and when to tackle this in the docs sprint-planning call tomorrow.

**Comment by rvagg** [comment]
:thumbsup: I'm keen to get this over the line, especially because it's so close. If I don't provide input earlier and you have some specific need for me just holler. I still have it on my list to try and get to this and figure out what can be done to make it happen.

**Comment by BigLep** [comment]
@johnnymatthews : @rvagg and I spoke.  We are good to merge this (after rebase) without syntax highlighting.  Can EngRes docs get it rebased/merged?  Thanks!

**Comment by rvagg** [comment]
@johnnymatthews I heard a rumour that doks may be on the way out for officially supported PL doc sites, is that true? If we move ahead here, are we in for further churn later on, or being stuck with an unsupported site in some form?

**Pull request closed without merging**